bindings/python/pymongoarrow/api.py

```diff
@@ -279,17 +292,30 @@ def write(collection, tabular):
     :Returns:
       An instance of :class:`result.ArrowWriteResult`.
     """
-
-    if isinstance(tabular, Table):
-        _validate_schema(tabular.schema.types)
-    elif isinstance(tabular, DataFrame):
-        _validate_schema(ArrowSchema.from_pandas(tabular).types)
     cur_offset = 0
     results = {
         "insertedCount": 0,
     }
-    tabular_gen = _tabular_generator(tabular)
     tab_size = len(tabular)
+    if isinstance(tabular, Table):
+        _validate_schema(tabular.schema.types)
+    elif isinstance(tabular, DataFrame):
+        _validate_schema(ArrowSchema.from_pandas(tabular).types)
+    elif (
+        isinstance(tabular, dict)
+        and len(tabular.values()) >= 1
+        and all([isinstance(i, ndarray) for i in tabular.values()])
+    ):
+        _validate_schema([i.dtype for i in tabular.values()])
+        tab_size = len(next(iter(tabular.values())))
+    else:
+        raise ValueError(
+            f"You passed an invalid tabular data object of type {type(tabular)} \n"
```

Comment:
"You passed an" is odd phrasing for an error message. Let's say:
`invalid tabular argument of type....`